### DISCUSSION TITLE
How to translate this input surrounding?

### DISCUSSION THREAD

**Discussion by xigoi** [nvim-mini/mini.nvim#131] [Q&A] [answered]
I see the custom surrounding system for `mini.surround` has changed. I have this input surrounding defined using the old system:

```lua
input = { find = "%b[]", extract = "^(%[[^%s%[%]]*%s*).*(%])$" },
```

How do I translate it to the new system? The guide in the documentation can't be applied because ane pattern for `find` doesn't contain the pattern for `replace`,

**Comment by echasnovski** [comment, marked as answer]
I am not sure why documented approach wouldn't work.

Does next setting behave not the same as this one?
```lua
input = { "%b[]", "^%[[^%s%[%]]*%s*().*()%]$" },
```

**Reply by echasnovski** [reply]
Oh, probably because one example shows those patterns merged together. It doesn't *need* to be, it was just an example.

Transition should literally just be:
- Remove `find = ` and `extract = `.
- Move `(` of left match to its right part (forming `()`).
- Move `)` of right match to its left part (also forming `()`).
